src/charm.py

```diff
@@ -4,27 +4,263 @@
 
 """Charm for MySQL."""
 
+import hashlib
 import logging
+import secrets
+import string
 
-from ops.charm import CharmBase
+from ops.charm import ActionEvent, CharmBase, RelationJoinedEvent
 from ops.main import main
-from ops.model import ActiveStatus
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
+from ops.pebble import Layer
+
+from mysqlsh_helpers import (
+    MySQL,
+    MySQLBootstrapInstanceError,
+    MySQLConfigureInstanceError,
+    MySQLConfigureMySQLUsersError,
+    MySQLCreateClusterError,
+    MySQLPatchDNSSearchesError,
+)
 
 logger = logging.getLogger(__name__)
 
+PASSWORD_LENGTH = 24
+PEER = "database-peers"
+CONFIGURED_FILE = "/var/lib/mysql/charmed"
+
+
+def generate_random_password(length: int) -> str:
+    """Randomly generate a string intended to be used as a password.
+
+    Args:
+        length: length of the randomly generated string to be returned
+    """
+    choices = string.ascii_letters + string.digits
+    return "".join([secrets.choice(choices) for i in range(length)])
+
+
+def generate_random_hash() -> str:
+    """Generate a hash based on a random string."""
+    random_characters = generate_random_password(10)
+    return hashlib.md5(random_characters.encode("utf-8")).hexdigest()
+
 
 class MySQLOperatorCharm(CharmBase):
     """Operator framework charm for MySQL."""
 
     def __init__(self, *args):
         super().__init__(*args)
+        self._mysqld_service = "mysqld"
```

Comment:
Since this is not changed, only accessed, it could be made a literal at the top. This is personal preference more than anything.

Reply:
Same here, the reasoning was what others were doing, but I may reconsider.

Reply:
Change as suggested